lessons/r/ggplot2/ggplot2.Rmd

```diff
@@ -0,0 +1,423 @@
+  
+  
+```{r}
+ggplot(diamonds, aes(x = carat, y = price)) +
+  geom_point(alpha = 0.5)
+```
+
+From the eye test, we can see some evidence of clusters in this scatter plots.
+It's possible that one of the categorical variables we have in the data could
+explain this pattern. Coloring the data points according to a categorical
+variable is an easy way to assess if there seems to be correlation. Let's try
+this with `cut`.
+
+```{r}
+ggplot(diamonds, aes(x = carat, y = price, color = cut)) +
+  geom_point(alpha = 0.5)
+```
+
+While some clustering is visible here, the pattern is not especially clear.
+Perhaps something else in our dataset might be a better predictor. Let's try
+`clarity` instead:
+
+```{r}
+ggplot(diamonds, aes(x = carat, y = price, color = clarity)) +
+geom_point(alpha = 0.5)
+```
+
+Turns out clarity appears to be a strong predictor here after all.  Given that
+IF1 is the best possible clarity measurement, we see that even 1-2 carat IF1
+diamonds fetch equivalent prices to 4-5 carat I1 (lowest clarity) diamonds.
+
+### Combining `dplyr` and `ggplot` - line plots
+
+What if we wanted to see how average carat varies by price?
+We'd first have to calculate what the mean carat value is for each price
+before plotting. Fortunately for us, we can bring back some of those `dplyr`
+skills from our previous lesson to do just that.
+
+```{r}
+diamonds %>% 
+  group_by(price) %>% 
+  summarise(mean_carat = mean(carat))
+```
+
+This transformed dataset can be piped right into a `ggplot` call, with which
+we can proceed to plot:
+
+```{r}
+diamonds %>% 
+  group_by(price) %>% 
+  summarise(mean_carat = mean(carat)) %>% 
+  ggplot(aes(x = price, y = mean_carat)) +
+  geom_line()
+```
+
+There's definitely a lot going on here -- let's just focus on the price range 0 - 5000:
+
+```{r}
+diamonds %>% 
+  group_by(price) %>% 
+  summarise(mean_carat = mean(carat)) %>% 
+  filter(price < 5000) %>% 
+  ggplot(aes(x = price, y = mean_carat)) +
+  geom_line()
+```
+
+
+### Histograms
+
+Similarly to how we've made line plots and scatter plots so far, histograms in `ggplot2`
+can simply be tacked on as another geom. The difference is that histograms do not
+take in a y aesthetic, since that'll be 'count' by definition:
+
+```{r}
+ggplot(diamonds, aes(x = carat)) +
+  geom_histogram()
+```
+
+The binning of a given histogram can be modified by either specifying a binwidth or a fixed number of bins:
+
+```{r}
+ggplot(diamonds, aes(x = carat)) +
+  geom_histogram(binwidth = 0.05)
+```
+
+### Bar plots with `dplyr` and `ggplot` - using grouped `summarise` operations
+
+Earlier, we saw that the `clarity` variable seemed to be associated with the carat of a given diamond.
+Let's investigate this further by calculating the mean carat for each value of `clarity`.
+This can be done using a grouped `summarise` operation via `dplyr`:
+
+```{r}
+diamonds %>% 
+  group_by(clarity) %>% 
+  summarise(mean_carat = mean(carat),
+            sd_carat = sd(carat))
```

Comment:
I would maybe do standard error instead of sd since it's more commonly plotted. This would require defining `n = n()` so that SE can be calculated as `se = sd / sqrt(n)`